### ISSUE TITLE
Release artifacts (models, dataset) on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [scxue/advantage_weighted_matching#1]
Hi @scxue 🤗

Niels here from the open-source team at Hugging Face. I discovered your work on Arxiv and was wondering whether you would like to submit it to [hf.co/papers](https://hf.co/papers) to improve its discoverability.If you are one of the authors, you can submit it at https://huggingface.co/papers/submit.

The paper page lets people discuss about your paper and lets them find artifacts about it (your models, datasets or demo for instance), you can also claim
the paper as yours which will show up on your public profile at HF, add Github and project page URLs.

I noticed your GitHub repository states "Coming Soon" for the code release of your Advantage Weighted Matching (AWM) method. It'd be great to make any associated checkpoints (e.g., AWM-optimized versions of Stable Diffusion 3.5 Medium or FLUX, if applicable) and any new datasets available on the 🤗 hub, to improve their discoverability/visibility once they are ready.
We can add tags so that people find them when filtering https://huggingface.co/models and https://huggingface.co/datasets.

## Uploading models

See here for a guide: https://huggingface.co/docs/hub/models-uploading.

In this case, we could leverage the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which adds `from_pretrained` and `push_to_hub` to any custom `nn.Module`. Alternatively, one can leverages the [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download#download-a-single-file) one-liner to download a checkpoint from the hub.

We encourage researchers to push each model checkpoint to a separate model repository, so that things like download stats also work. We can then also link the checkpoints to the paper page.

## Uploading dataset

Would be awesome to make any new datasets available on 🤗 , so that people can do:

```python
from datasets import load_dataset

dataset = load_dataset("your-hf-org-or-username/your-dataset")
```
See here for a guide: https://huggingface.co/docs/datasets/loading.

Besides that, there's the [dataset viewer](https://huggingface.co/docs/hub/en/datasets-viewer) which allows people to quickly explore the first few rows of the data in the browser.

Let me know if you're interested/need any help regarding this, once your code and artifacts are publicly available!

Cheers,

Niels
ML Engineer @ HF 🤗

**Comment by scxue** [comment]
Hi Niels,

Thanks for reaching out and for the helpful pointers! I’ve submitted the paper to hf.co/papers and claimed it on my profile. Our code and model checkpoints (e.g., AWM-tuned SD3.5-M / FLUX) are currently going through company review.

Appreciate the offer—I'll ping you if I need help with tags or setup.

Best,
Shuchen
